EIPS/eip-1193.md

```diff
@@ -108,7 +108,7 @@ The event emits with `accounts`, an array of the accounts' addresses.
 ## Examples
 
 ```js
-const ethereum = window.ethereum;
+const ethereum = new EthereumProvider();
```

Comment:
If the goal of this change is to decouple the provider from the environment, why introduce a new global variable here? Is this not intended to be a global variable? Maybe it would be worth noting that this `EthereumProvider` class can be provided in an arbitrary way, dependent on its environment. This would allow solutions like Micah likes to suggest, where nothing is injected initially, but this class could be returned in response to a `PostMessage`.

Reply:
Yes, this was a follow to a discussion on Ethereum Provider Ring's discord where we discussed Micah's proposal as well.

In short, we want to abstract the transport implementation, narrowing the scope of the EIP-1193 to simply the API interface, .

I agree that the way I made the changes opens up more questions to how this Transport class is constructed which itself would require an extra EIP. 

I would suggest then that we remove this L111 and also revert the changes I made to the Sample Class Implementation.

Reply:
However I would like to challenged the fact that this Transport abstraction is a "if it's not broke, don't fix it" situation. 

The way we injected web3/ethereum providers for Dapps is something that needs to be changed IMHO. Not only as Micah pointed out, a bad browser practice, but also it completely skews the way Dapps are developed today to handle different providers.

Reply:
The provider’s inner transport is a distinct issue from the mechanism by which a context gets access to a provider. I still don’t see any reason to specify Transport, even if we popularize Micah’s postMessage injection, which supplies a provider regardless of its transport.

Reply:
Agreed, it's out of scope of this EIP.